ch_tools/monrun_checks/ch_backup.py

```diff
@@ -288,6 +292,23 @@ def _merge_results(*results: Result) -> Result:
     return merged_result
 
 
+def _is_backup_failed_by_userfault_error(backups: List[Dict]) -> bool:
+    failed_backup = None
+    for i, backup in enumerate(backups):
+        state = backup["state"]
+        if state == "creating" and i == 0:
+            continue
+        if state == "failed":
+            failed_backup = backup
+        break
+
+    if not failed_backup:
+        return False
+
+    exception = failed_backup.get("exception", "")
+    return any(value in exception for value in USERFAULT_ERRORS)
```

Comment:
```suggestion
    return _is_userfault_exception(exception)
```

Reply:
After refactoring `_is_userfault_exception` looks redundant. Removed it. Also updated tests.